lib/wsapp.js

```diff
@@ -12,15 +12,81 @@ var MessageBroker = require('./messagebroker');
 
 log.level = 'debug';
 
-var WsApp = function() {};
-
-WsApp.prototype._unauthorized = function(socket) {
-  socket.emit('unauthorized');
-  socket.disconnect();
+var WsApp = function() {
+    this.connectedWallets = {};
+    this.authorizedWallets = {};
 };
 
+function keysToArray(dict){
+  var ret = [];
+  for (var key in dict)
+    ret.push(key);
+  return ret;
+}
+
 WsApp.prototype._handleNotification = function(notification) {
-  this.io.to(notification.walletId).emit('notification', notification);
+  var namespace = this.io.to(notification.walletId);
+  var to = this.authorizedWallets[notification.walletId] || {};
+  console.log('Notification: ', notification);
+  namespace.emit('multimessage', {to: keysToArray(to), message: 'notification', payload: notification});
```

Comment:
What happens if the list of recipients is empty? Isn't it better to test this condition before emitting?